oop/homework.py

```diff
@@ -0,0 +1,409 @@
+                                     for k, v in product_dict.items() if product == k][0]
+        return self.saturation_level
+
+    def _set_average_speed(self):
+        if self.age <= 5:
+            self.average_speed = 90
+        if 5 < self.age <= 15:
+            self.average_speed = 75
+        if self.age > 15:
+            self.average_speed = 40
+        return self.average_speed
+
+    def __str__(self):
+        return f"""
+    Current cheetah is {self.age} years old.
+        """
+
+
+class Wall:
+    """
+    * Implement class Wall which receives such parameters: width and height
+
+    * Implement method wall_square which return result of simple square formula of rectangle
+
+    * Implement method number_of_rolls_of_wallpaper which receives such parameters: roll_width_m, roll_length_m
+      (_m in the parameters name means meters) return number of rolls of wallpaper
+
+      Example:
+          count of lines in roll eq roll length in meters divide height of the wall (use rounding down)
+          count of lines eq width of the wall divide roll width in meters
+          number of rolls of wallpaper eq count of lines divide  count of lines in roll
+    """
+
+    def __init__(self, width, height):
+        self.width = width
+        self.height = height
+
+    def wall_square(self):
+        return self.width * self.height
+
+    def number_of_rolls_of_wallpaper(self, roll_width_m, roll_length_m):
+        lines_in_roll = m.floor(roll_length_m / self.height)
+        lines_for_wall = m.floor(self.width / roll_width_m)
+        return lines_for_wall / lines_in_roll
+
+
+class Roof:
+    """
+        * Implement class Roof which receives such parameters: width, height and roof_type
+
+        * Implement method roof_square that returns square of the roof
+          if roof_type eq "gable" the roof square if simple rectangle square formula multiplied 2
+          if roof_type eq "single-pitch" the roof square if simple rectangle square formula
+          if other roof_type raise ValueError like this "Sorry there is only two types of roofs"
+
+    """
+
+    def __init__(self, width, height, roof_type):
+        self.width = width
+        self.height = height
+        self.roof_type = roof_type
+
+    def roof_square(self):
+        if self.roof_type == "gable":
+            return self.width * self.height * 2
+        if self.roof_type == "single-pitch":
+            return self.width * self.height
+        else:
+            raise ValueError("Sorry there is only two types of roofs")
+
+
+class Window:
+    """
+       * Implement class Window which receives such parameters: width and height
+
+       * Implement method window_square which return result of simple square formula of rectangle
+
+    """
+
+    def __init__(self, width, height):
+        self.width = width
+        self.height = height
+
+    def window_square(self):
+        return self.width * self.height
```

Comment:
What happens when width or height will be 0?

Reply:
window_square = 0
:)